src/ui/actions/timeline.ts

```diff
@@ -186,6 +208,22 @@ export function setZoomRegion(region: ZoomRegion): SetZoomRegionAction {
   return { type: "set_zoom", region };
 }
 
+function updateUrl({
+  point,
+  time,
+  hasFrames,
+}: {
+  point: ExecutionPoint;
+  time: number;
+  hasFrames: boolean;
+}) {
+  const url = new URL(window.location.toString());
+  url.searchParams.set("point", point);
+  url.searchParams.set("time", `${time}`);
+  url.searchParams.set("hasFrames", `${hasFrames}`);
+  window.history.pushState({}, "", url.toString());
```

Comment:
we're using `react-router`, is it safe to mix that with direct access to `window.history`?

Reply:
I think so... I looked into a tighter integration briefly, but nothing was obvious...